Subject: [PATCH]  Fix Bad activity token

 ActivityManager: Bad activity token: android.os.BinderProxy@9fa357
 ActivityManager: java.lang.ClassCastException: android.os.BinderProxy cannot be cast to com.android.server.am.ActivityRecord$Token
 ActivityManager: 	at com.android.server.am.ActivityRecord.forTokenLocked(ActivityRecord.java:597)
 ActivityManager: 	at com.android.server.am.ActivityRecord.isInStackLocked(ActivityRecord.java:1373)

diff --git a/services/core/java/com/android/server/am/ActivityRecord.java b/services/core/java/com/android/server/am/ActivityRecord.java
index 1a0e45e..4dc5764 100755
--- a/services/core/java/com/android/server/am/ActivityRecord.java
+++ b/services/core/java/com/android/server/am/ActivityRecord.java
@@ -590,6 +590,9 @@
     }
 
     static ActivityRecord forTokenLocked(IBinder token) {
+        if (token == null) {
+            return null;
+        }
         try {
             return Token.tokenToActivityRecordLocked((Token)token);
         } catch (ClassCastException e) {
